### ISSUE TITLE
minimizing_the_negative_log_likelihood_in_english

### ISSUE THREAD

**Issue by sweaterr** [sweaterr/deep-learning-readinglist#83]
http://willwolf.io/2017/05/18/minimizing_the_negative_log_likelihood_in_english/

**Comment by sweaterr** [comment]
대략 말하면서, 기계 학습 여행은 Kaggle에서 시작되었습니다. "최적화 된 데이터, 모델 (즉, 추정량) 및 손실 함수가 있습니다."내가 배웠습니다. "회귀 모델은 연속 값의 실수를 예측하며 분류 모델은 '빨강, 녹색, 파랑'을 예측합니다. 일반적으로 전자는 평균 제곱 오류 또는 평균 절대 오류를 사용하며, 후자는 교차 엔트로피 손실을 사용합니다. 확률 적 그라디언트 디센트는 모델의 매개 변수를 업데이트하여 이러한 손실을 줄입니다. 또한,이 모델에 맞게 sklearn을 가져 오기만하면됩니다.

**Comment by sweaterr** [comment]
위와 관련된 손재주는 적어도 기술적 인 입장에서 보면 데이터 과학자로서 고용과 영향 모두에 충분합니다. 업계에서는 평범한 예측 및 추론 문제 - 예를 들어 바이너리 채움, 신용 점수 계산, 제품 권장 및 A / B 테스트 -을 상용 알고리즘 및 능숙 데이터 과학자와 쉽게 비교하여 회사 수익에 상당한 향상을 가져옵니다 . 진공 상태에서 나는 이것이 괜찮다고 생각한다. 승리하는 운전자는 차를 만드는 법을 알 필요가 없다. 분명히, 나는이 사람을 전에 만 봤다.

**Comment by sweaterr** [comment]
Once fluid with "scikit-learn fit and predict," I turned to statistics. I was always aware that the two were related, yet figured them ultimately parallel sub-fields of my job. With the former, I build classification models; with the latter, I infer signup counts with the Poisson distribution and MCMC — right?
일단 "scikit-learn fit과 예측"을 가진 유체는 통계로 바뀌 었습니다. 나는이 두 가지가 관련되어 있다는 것을 항상 알고 있었지만 궁극적으로는 내 직업의 병렬 하위 필드라고 생각했다. 전자는 분류 모델을 구축합니다. 후자와 함께, 나는 포아송 분포와 MCMC의 가입 수를 추측한다 - 맞습니까?

**Comment by sweaterr** [comment]
Before long, I dove deeper into machine learning — reading textbooks, papers and source code and writing this blog. Therein, I began to come across terms I didn't understand used to describe the things that I did. "I understand what the categorical cross-entropy loss is, what it does and how it's defined," for example: "why are you calling it the negative log-likelihood?"
오래 전에 저는 교과서, 논문 및 소스 코드 읽기 및이 블로그 작성과 같은 기계 학습에 대해 깊이 파고 들었습니다. 거기에서, 나는 내가 한 일들을 묘사하는 데 익숙하지 않은 조건들을 만나기 시작했다. "범주 적 엔트로피 손실이 무엇인지, 그것이 무엇이며 어떻게 정의되는지 이해합니다."예를 들면 : "왜 당신은 그것을 음의 대수 우도라고 부릅니까?"

**Comment by sweaterr** [comment]
The goal of this post is to take three models we know, love, and know how to use and explain what's really going on underneath the hood. I will assume the reader is familiar with concepts in both machine learning and statistics, and comes in search of a deeper understanding of the connections therein. There will be math — but only as much as necessary. Most of the derivations can be skipped without consequence.
이 게시물의 목표는 우리가 알고 있고, 사랑하고, 사용 방법을 알고 후드 아래에서 실제로 진행되는 것을 설명하는 3 가지 모델을 얻는 것입니다. 나는 독자가 기계 학습과 통계학에 대한 개념에 익숙하다고 가정하고 거기에서의 연관성에 대해 더 깊이 이해할 것이다. 수학이있을 것입니다 - 그러나 필요한만큼만. 유도의 대부분은 결과없이 건너 뛸 수 있습니다.

**Comment by sweaterr** [comment]
When deploying a predictive model in a production setting, it is generally in our best interest to import sklearn, i.e. use a model that someone else has built. This is something we already know how to do. As such, this post will start and end here: your head is currently above water; we're going to dive into the pool, touch the bottom, then work our way back to the surface. Lemmas will be written in bold.

프로덕션 환경에서 예측 모델을 배포 할 때 일반적으로 다른 사람이 만든 모델을 사용하여 sklearn을 가져 오는 것이 가장 좋습니다. 이것은 우리가 어떻게해야하는지 이미 알고있는 것입니다. 따라서이 게시물은 여기에서 시작되고 끝납니다 : 당신의 머리는 현재 물 위에 있습니다. 우리는 웅덩이 속으로 잠수하고, 바닥을 만질 것이고, 그 길로 다시 일할 것입니다. Lemmas는 굵게 표시됩니다.

**Comment by sweaterr** [comment]
I define a random variable as "a thing that can take on a bunch of different values."
무작위 변수를 "다른 값을 가질 수있는 것"으로 정의합니다.

A probability distribution is a lookup table for the likelihood of observing each unique value of a random variable
확률 분포는 확률 변수의 각 고유 값을 관찰 할 가능성에 대한 룩업 테이블입니다

Entropy quantifies the number of ways we can reach a given outcome.
엔트로피는 주어진 결과에 도달 할 수있는 방법의 수를 정량화합니다.

**Comment by sweaterr** [comment]
요약하면, 
끝단 레이어의 예측 모델이 linear regression, logistic reg, softmax 인경우, 전부 exponential family 꼴로 변환할 수 있고, 이들의 negative log likelohood의  mle 하는 파라메터를 구하면, cross entropy 꼴이 나온다
